packages/react-dom-bindings/src/server/ReactDOMServerFormatConfig.js

```diff
@@ -2001,65 +2034,172 @@ export function writeEndSegment(
 //   }
 // }
 //
-// function completeBoundary(suspenseBoundaryID, contentID) {
-//   // Find the fallback's first element.
-//   const suspenseIdNode = document.getElementById(suspenseBoundaryID);
-//   const contentNode = document.getElementById(contentID);
-//   // We'll detach the content node so that regardless of what happens next we don't leave in the tree.
-//   // This might also help by not causing recalcing each time we move a child from here to the target.
-//   contentNode.parentNode.removeChild(contentNode);
-//   if (!suspenseIdNode) {
-//     // The user must have already navigated away from this tree.
-//     // E.g. because the parent was hydrated. That's fine there's nothing to do
-//     // but we have to make sure that we already deleted the container node.
-//     return;
+// const insertStyles = (function() {
+//   const resourceMap = new Map();
+//   const precedences = new Map();
+//   // We omit rel stylesheet because only stylesheets should have data-prec attribute
+//   // and we can concievably use another kind of link to act as a placeholder for a
+//   // precedence that does not yet have any resources.
+//   let lastResource, node;
+//   let thisDocument = document;
```

Comment:
Why `thisDocument` instead of just `document`? It can't change, can it?

Reply:
bytes